src/operations/estimated_document_count.ts

```diff
@@ -1,62 +1,63 @@
-import { Aspect, defineAspects, Hint } from './operation';
+import { Aspect, defineAspects } from './operation';
 import { CommandOperation, CommandOperationOptions } from './command';
-import type { Callback } from '../utils';
+import { Callback, maxWireVersion } from '../utils';
 import type { Document } from '../bson';
 import type { Server } from '../sdam/server';
 import type { Collection } from '../collection';
 import type { ClientSession } from '../sessions';
+import type { MongoError } from '../error';
 
 /** @public */
 export interface EstimatedDocumentCountOptions extends CommandOperationOptions {
-  skip?: number;
-  limit?: number;
-  hint?: Hint;
+  /**
+   * The maximum amount of time to allow the operation to run.
+   *
+   * This option is sent only if the caller explicitly provides a value. The default is to not send a value.
+   */
+  maxTimeMS?: number;
 }
 
 /** @internal */
 export class EstimatedDocumentCountOperation extends CommandOperation<number> {
   options: EstimatedDocumentCountOptions;
   collectionName: string;
-  query?: Document;
 
-  constructor(collection: Collection, options: EstimatedDocumentCountOptions);
-  constructor(collection: Collection, query: Document, options: EstimatedDocumentCountOptions);
-  constructor(
-    collection: Collection,
-    query?: Document | EstimatedDocumentCountOptions,
-    options?: EstimatedDocumentCountOptions
-  ) {
+  constructor(collection: Collection, options?: EstimatedDocumentCountOptions) {
```

Comment:
```suggestion
  constructor(collection: Collection, options: EstimatedDocumentCountOptions = {}) {
```
Just a preference to use this, and then drop the if stmt below, feel free to disagree ofc.